### PR TITLE
Define some structure for lifetime values 

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Here is a description of how to encode lifetime values. We intend to include the types and values defined in this pull request with the same semantics in an upcoming release of the PSA Cryptography API specification.

This proposal is backward-compatible with the current values for volatile keys (0) and persistent keys (1). Platforms with a secure element (i.e. a cryptoprocessor that can process keys which are not accessible to the PSA crypto core) should use the lifetime 0x1ff (whether the key material is stored inside the secure element, or in generic PSA storage in a form wrapped by the secure element).

This pull request defines the lifetime structure, but does not yet implement any new feature. A future pull request will adapt the (currently experimental) secure element support code accordingly.

This stems from an internal discussion ([private link](https://github.com/ARMmbed/psa-crypto-api/issues/207#issuecomment-582474856)). This pull request is a continuation of https://github.com/ARMmbed/mbed-crypto/pull/358. The first commit here is the final state of that mbed-crypto PR (the history there wasn't worth carrying over).

Status: API approved. Now needs to be implemented (treat all volatile values as such, even if they're in a secure element).

### CODE REVIEW

**Comment by mpg** [comment]
I looked at the reported failure in the pr-merge job and it looks like a temporary glitch in the infrastructure - failed to clone the repo - not a problem with this PR.

File: include/psa/crypto_types.h [hunk 1]
```diff
@@ -108,18 +108,117 @@ typedef uint32_t psa_algorithm_t;
...
+ *   contains the key metadata.
+ * - \c 1: primary secure element.
+ *   Implementations should support this value if there is a secure element
+ *   attached to the operating environment.
+ *   As a guideline, secure elements may provide higher resistance against
+ *   side channel and physical attacks than the primary local storage, but may
+ *   have restrictions on supported key types, sizes, policies and operations
+ *   and may have different performance characteristics.
+ * - \c 2-0x7fffff: other locations defined by a PSA specification.
+ *   The PSA Cryptography does not currently assign any meaning to these
```
Comment (yanesca) [inline include/psa/crypto_types.h:210]:
```suggestion
 *   The PSA Cryptography API does not currently assign any meaning to these
```

**Comment by mpg** [comment]
Does this need a ChangeLog entry? I'm not sure.

**Comment by gilles-peskine-arm** [comment]
So far new PSA features haven't been getting a changelog entry, but we've started to write changelog entries for behavior change. In any case, this is documentation only so far, the new feature isn't implemented.

The documentation does indicate that certain lifetime values are now considered volatile, but the code doesn't treat them as such. (This is not a behavior change because the volatility of these values was previously unspecified.) So, annoyingly, we shouldn't merge this yet.

**Pull request merged**